src/test/java/part1/exercise/StreamsExercise2.java

```diff
@@ -74,7 +74,10 @@ public void indexByFirstEmployer() {
         }
 
         final Map<String, List<Person>> employeesIndex = getEmployees().stream()
-                .map(e -> new PersonEmployerPair(e.getPerson(), e.getJobHistory().get(0).getEmployer()))
+                .map(e -> e.getJobHistory().stream().findFirst()

```

Comment:
Use `flatMap(...findFirst().map(...).map(Stream::of).orElseGet(Stream::empty))` or `flatMap(...limit(1))`